luautils/fs.go

```diff
@@ -0,0 +1,106 @@
+package luautils
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	lua "github.com/yuin/gopher-lua"
+)
+
+var basePath string
+
+// SetBasePath sets the base path for file operations
+func SetBasePath(path string) {
+	basePath = path
+}
+
+// RegisterFSModule registers the fs module functions
+func RegisterFSModule(L *lua.LState) {
+	mod := L.RegisterModule("fs", map[string]lua.LGFunction{
+		"read": fsRead,
+		"walk": fsWalk,
+	})
+	L.Push(mod)
+}
+
+func fsRead(L *lua.LState) int {
+	filePath := L.CheckString(1)
+
+	// Validate and clean the path
+	cleanPath, err := validatePath(filePath)
+	if err != nil {
+		L.Push(lua.LNil)
+		L.Push(lua.LString(err.Error()))
+		return 2
+	}
+
+	content, err := os.ReadFile(cleanPath)
+	if err != nil {
+		L.Push(lua.LNil)
+		L.Push(lua.LString(err.Error()))
+		return 2
+	}
+
+	L.Push(lua.LString(string(content)))
+	return 1
```

Comment:
is it easy to handle this if it returns different sets of return values?

Reply:
what do you mean exactly? Are you asking in general how to manage returned values?